content.js

```diff
@@ -143,6 +144,37 @@ function removeChatFromLocalStorage(href) {
 }
 
 function removeSavedChat(href, savedChatsList) {
```

Comment:
Also this function is really long now, can you refactor this into a few a few smaller functions

Reply:
### Done! 

![Screenshot 2024-12-28 074434](https://github.com/user-attachments/assets/b949b1d0-07be-40bf-91ba-1d1c1553281f)

### Changes:

- createConfirmationDialog: Added a dialog to confirm chat deletion, displaying the chat name.
- handleDelete:  Handles the deletion of the chat only after confirmation.
- attachDialogListeners:  Manages the dialog’s event listeners (e.g., close and delete actions).
- removeSavedChat:  Triggers the confirmation dialog and proceeds with deletion based on user action.

Feel free to share any further bugs or suggestions for improvement. I hope this resolves the issue for now.
Thanks!